test/HydratorDeprecationTest.php

```diff
@@ -23,4 +23,15 @@ public function testPassingHydratorExtendingStdlibAbstractHydratorToTypehintedMe
         $hydratorInjected->setHydrator($hydrator);
         $this->assertSame($hydrator, $hydratorInjected->hydrator);
     }
+
+    public function testDeprecatedHydratorInterfaceIsAcceptedByMethodsTypehintedWithNewInterface()
+    {
+        $hydratorInjected = new TestAsset\HydratorInjectedObjectUsingDeprecatedInterfaceHydrator();
```

Comment:
I think it's more simple build custom mocks with Prophet implementing the desired interfaces and then use multiple assertInstanceOf for to verify the desired hierarchy tree works.

PD: I know this test is a copy paste from the above test.